### ISSUE TITLE
macOS release?

### ISSUE THREAD

**Issue by jucor** [werman/noise-suppression-for-voice#101]
I've used Noisesupression on Windows and it's been fantastic, thank you very very much!! 
Just moved to macOS (m1) -- is there anyway to use it there, please? I don't find it in the releases, but it would be awesome.

**Comment by werman** [comment]
I don't have where to build the macOS release, you could compile it yourself - https://github.com/werman/noise-suppression-for-voice/issues/6 (there is also a compiled binary there but it was compiled not by me)

**Comment by pyromind** [comment]
Could somebody compile it for M1 Macs please?

**Comment by richtong** [comment]
I am trying to compile this for an M1 Mac, at least on MacOS Monterey it generate an error 

```
 40%] Building CXX object CMakeFiles/rnnoise_vst.dir/vst2.x/audioeffect.cpp.o
/Users/rich/ws/git/src/extern/noise-suppression-for-voice/src/vst_plugin/vst2.x/audioeffect.cpp:530:20: error: non-constant-expression cannot be narrowed from type 'int' to 'char' in initializer list [-Wc++11-narrowing]
                        char temp[2] = {'0' + (char)digit, '\0'};
```

LOoks like a problem with the version of the c compiler

**Comment by mischultheis** [comment]
I have the same issue on M1/Apple Silicon and MacOS Monterey.

**Comment by jucor** [comment]
Looks like either we have to patch vst2.4 plugin (nope!) to be compatible with C++11 restrictions on narrowing conversions from non-constants in static initializers, *or* we need to tell to CMake to ignore the `c++11-narrowing` warning -- which is triggered, and converted to an error because I suspect somewhere there must be `-Werror` flag?

**Comment by jucor** [comment]
I'm no CMAKE expert, but I suspect some of the modifications from this commit https://github.com/werman/noise-suppression-for-voice/commit/61bee35dd81465ce3982193365d0f6dac9f8f6c4 could probably be adapted to this file https://github.com/werman/noise-suppression-for-voice/blob/6466b34d57e7b6017baf16b176c6b897d077d071/src/vst_plugin/CMakeLists.txt 
Anyone more Cmake-fluent than I am?

**Comment by jucor** [comment]
Looks like other nice folks have been bit before: https://github.com/surge-synthesizer/surge/issues/201

**Comment by jucor** [comment]
Mmm, looks like we somehow need to tell CMAKE to  pass `-Wno-c++11-narrowing -std=c++17` to `clang`. Any idea how to do that?

**Comment by jucor** [comment]
@richtong Could you try with 
`cmake -D CMAKE_CXX_FLAGS=-Wno-c++11-narrowing` (in addition to the other command line parameters you normally give Cmake), please? If we're lucky that might just work... 🤞

**Comment by baconpaul** [comment]
change the line to

```
  char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};
```

and one imagines it will work without having to turn on those somewhat scary flags.

**Comment by rafaeljustinox** [comment]
And after compiling? How can I "load" de plugin on Mac?

**Comment by weitheng** [comment]
> change the line to
> 
> ```
>   char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};
> ```
> 
> and one imagines it will work without having to turn on those somewhat scary flags.

This works! But I have no idea what do I do after compiling it. Am I supposed to drag some file (vst file?) to /Library/Audio/Plug-Ins? If so, where can I find the file?

**Comment by werman** [comment]
> > > change the line to
> > > ```
> > >   char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > and one imagines it will work without having to turn on those somewhat scary flags.
> > 
> > 
> > This works! But I have no idea what do I do after compiling it. Am I supposed to drag some file (vst file?) to /Library/Audio/Plug-Ins? If so, where can I find the file?
> 
> Same here, succesfully compiled but then what? :)

What do you want to do with the plugin? I mean, you have some app which you want to use the plugin in?

**Comment by nank1ro** [comment]
> > > > change the line to
> > > > ```
> > > >   char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > and one imagines it will work without having to turn on those somewhat scary flags.
> > > 
> > > 
> > > This works! But I have no idea what do I do after compiling it. Am I supposed to drag some file (vst file?) to /Library/Audio/Plug-Ins? If so, where can I find the file?
> > 
> > 
> > Same here, succesfully compiled but then what? :)
> 
> What do you want to do with the plugin? I mean, you have some app which you want to use the plugin in?

Is it possibile to use it to reduce the noise for calls with Google Meet (using the browser)? I have too an M1 Mac

**Comment by UmutAlihan** [comment]
> > > > > change the line to
> > > > > ```
> > > > >   char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > and one imagines it will work without having to turn on those somewhat scary flags.
> > > > 
> > > > 
> > > > This works! But I have no idea what do I do after compiling it. Am I supposed to drag some file (vst file?) to /Library/Audio/Plug-Ins? If so, where can I find the file?
> > > 
> > > 
> > > Same here, succesfully compiled but then what? :)
> > 
> > 
> > What do you want to do with the plugin? I mean, you have some app which you want to use the plugin in?
> 
> Is it possibile to use it to reduce the noise for calls with Google Meet (using the browser) or Slack? I have too an M1 Mac

extactly I was also planning to use with webex or zoom-like applications to cancel background noise

**Comment by nank1ro** [comment]
> Hello all! Just wanted to say, appreciate all the efforts that people in this thread have made toward compilation on an M1 Mac and Monterey! (As I am in the same boat 😂)
> 
> Would anyone have the final compiled binary?
> 
> @werman If you were provided the binary, is that something you'd be able to place into Releases for those (like me) that are not as familiar or knowledgeable in terminal and compiling space?
> 
> Thanks again all! 🙏

You can simply follow these instructions https://github.com/werman/noise-suppression-for-voice/issues/6#issuecomment-462086803
If you don't have cmake, you can simply run
`brew install cmake`
If you don't even have brew, simply follow the [instructions provided here ](https://brew.sh/)

**Comment by maosante** [comment]
Hey guys, I've tried to compile the VST with no success, error bellow

```
/Users/maosante/Downloads/Rnnoise/src/vst_plugin/vst2.x/audioeffect.cpp:512:20: error: non-constant-expression cannot be narrowed from type 'int' to 'char' in initializer list [-Wc++11-narrowing]
                        char temp[2] = {'0' + (char)digit, '\0'};
                                        ^~~~~~~~~~~~~~~~~
/Users/maosante/Downloads/Rnnoise/src/vst_plugin/vst2.x/audioeffect.cpp:512:20: note: insert an explicit cast to silence this issue
                        char temp[2] = {'0' + (char)digit, '\0'};
                                        ^~~~~~~~~~~~~~~~~
                                        static_cast<char>( )
1 error generated.
make[2]: *** [src/vst_plugin/CMakeFiles/rnnoise_vst.dir/vst2.x/audioeffect.cpp.o] Error 1
make[1]: *** [src/vst_plugin/CMakeFiles/rnnoise_vst.dir/all] Error 2
make: *** [all] Error 2
```

I'm trying to use the VST in Ableton Live or WavesLab. I feel this noise suppressor is much better that other fancy "noise suppressors" out there. I found it in OBS and want to use it in other programs. Hope you guys can help me achieve this goal.

I couldn't applied this edit from above because I don't even know where to change that line.

  char temp[2] = {static_cast<char>('0' + (char)digit), '\0'};

Thanks in advance for your help!

**Comment by werman** [comment]
I'm working on a new release and I hope to add macOS binaries this time.

**Comment by maosante** [comment]
> I'm working on a new release and I hope to add macOS binaries this time.

Sounds great! I'll be looking forward to this release. 👍🏼 Big thanks.

**Comment by jucor** [comment]
Awesome, thank you !

**Comment by pyromind** [comment]
That is great news! Do you know when you would publish the new release incl the MacOS binaries?

**Comment by werman** [comment]
> That is great news! Do you know when you would publish the new release incl the MacOS binaries?

I just pushed my latest changes to https://github.com/werman/noise-suppression-for-voice/commits/develop, alongside many changes now builds are done in CI.

There are macOS builds there, however I don't have means to test them. You could find the binaries at https://github.com/werman/noise-suppression-for-voice/actions/runs/2563715931 and I would highly appreciate if someone could test the VST2 plugin. 

(Now I'm using JUCE for plugins so I could also build VST3 AU AUv3 plugins, so if you want to test them instead - I'll build the binaries)

**Comment by werman** [comment]
Here is a full build with all plugin types: https://github.com/werman/noise-suppression-for-voice/actions/runs/2563748155

**Comment by maosante** [comment]
VST3 and Component works just fine on macOS Big Sur 11.5.2 Thank You! Here's the GUI. Controls also worked  fine, yet I would like to know the function of VAD Grace Period and Retroactive VAD Grace knobs, by testing them it seems to be like an attack a release controls. Correct me if I'm wrong. By the way great work!
<img width="399" alt="Captura de Pantalla 2022-06-26 a la(s) 12 25 31 p  m" src="https://user-images.githubusercontent.com/95062236/175826462-b9daf2a9-06a9-4d4c-a375-5b6dc21f9e56.png">

**Comment by werman** [comment]
@maosante Thanks for Testing!

- "VAD Grace Period" - Duration for which output won't be muted after the last time voice was detected, fixes cut offs at the end of the sentences;
- "Retroactive VAD Grace" - The same, but before the voice is detected; fixes cut offs at the start of the sentences. (There is also a bug that the latency is not reduced back when you reduce the value, until plugin is reloaded)

Though the "blocks" measurement is not intuitive so I probably rename it into milliseconds and add some explanation.

Btw, which app did you use to test?

**Comment by maosante** [comment]
> @maosante Thanks for testing!
> 
> * "VAD Grace Period" - Duration for which output won't be muted after the last time voice was detected, fixes cut offs at the end of the sentences;
> * "Retroactive VAD Grace" - The same, but before the voice is detected; fixes cut offs at the start of the sentences. (There is also a bug that the latency is not reduced back when you reduce the value, until plugin is reloaded)
> 
> Though the "blocks" measurement is not intuitive so I probably rename it into milliseconds and add some explanation.
> 
> Btw, which app did you use to test?

I test it on Ableton Live 10 and Wave Lab LE 11. In both the plugin worked well with no issues at all.

Thanks for the controls explanation! 👍🏼 Surely the ms tag can be added instead of blocks. Hope to to keep on testing upcoming versions, yet this one does the job. Great work.

**Comment by werman** [comment]
Released v1.01, with macOS binaries.
